src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -175,20 +237,62 @@ static void Main()
             Assert.Null(model.GetSymbolInfo(def).Symbol);
-            Assert.False(model.GetConstantValue(def).HasValue);
+            Assert.True(model.GetConstantValue(def).HasValue);
             Assert.False(model.GetConversion(def).IsNullLiteral);
+            Assert.True(model.GetConversion(def).IsDefaultLiteral);
 
             var nullSyntax = nodes.OfType<LiteralExpressionSyntax>().ElementAt(1);
             Assert.Equal("null", nullSyntax.ToString());
             Assert.Null(model.GetTypeInfo(nullSyntax).Type);
-            Assert.Null(model.GetTypeInfo(nullSyntax).ConvertedType); // Should be given a type. Follow-up issue: https://github.com/dotnet/roslyn/issues/18609
+            Assert.Equal("System.Object", model.GetTypeInfo(nullSyntax).ConvertedType.ToTestDisplayString());
             Assert.Null(model.GetSymbolInfo(nullSyntax).Symbol);
         }
 
-        [Fact]
+        [Fact, WorkItem(35684, "https://github.com/dotnet/roslyn/issues/35684")]
+        public void ComparisonWithUnconstrainedGenericType()
+        {
+            string source = @"
+class C
+{
+    static bool M<T>(T x = default)
+    {
+        return x == default // 1
+            && x == default(T); // 2
+    }
+}
+";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular7_1);
+            comp.VerifyDiagnostics(
+                // (6,16): error CS0019: Operator '==' cannot be applied to operands of type 'T' and 'default'
+                //         return x == default // 1
+                Diagnostic(ErrorCode.ERR_BadBinaryOps, "x == default").WithArguments("==", "T", "default").WithLocation(6, 16),
+                // (7,16): error CS0019: Operator '==' cannot be applied to operands of type 'T' and 'T'
+                //             && x == default(T); // 2
+                Diagnostic(ErrorCode.ERR_BadBinaryOps, "x == default(T)").WithArguments("==", "T", "T").WithLocation(7, 16)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+
+            var default1 = nodes.OfType<LiteralExpressionSyntax>().ElementAt(0);
+            Assert.Equal("default", default1.ToString());
+            Assert.Equal("T", model.GetTypeInfo(default1).Type.ToTestDisplayString());
+            Assert.Equal("T", model.GetTypeInfo(default1).ConvertedType.ToTestDisplayString());
+            Assert.True(model.GetConversion(default1).IsDefaultLiteral);
+
+            var default2 = nodes.OfType<LiteralExpressionSyntax>().ElementAt(1);
+            Assert.Equal("default", default2.ToString());
+            Assert.Equal("?", model.GetTypeInfo(default2).Type.ToTestDisplayString());
+            Assert.Equal("?", model.GetTypeInfo(default2).ConvertedType.ToTestDisplayString());
+            Assert.False(model.GetConversion(default2).IsDefaultLiteral);
```

Comment:
is there no conversion occurring here? is it worthwhile to assert that specifically?